docs/src/proposals/fee_transaction_priority.md

```diff
@@ -47,111 +47,128 @@ locking complexity.
 
 #### Components of the `Scheduler`:
 
-1. `default_transaction_queue` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions.
-The priority in the heap is the additional fee of the transaction. Transactions are added to this queue
-from sigverify before the leader slot begins.
-2. `all_transaction_queues` - A `VecDeque<BinaryHeap<Transaction>>` that tracks all pending queues of work.
-Some pending queues have higher priority than others (as will be explained later in the `Handling Completion Signals from BankingStage Threads` section below). The list is ordered in priority from highest to lowest priority. On
-initialization, `all_transaction_queues[0] = default_transaction_queue`.
-3. `locked_accounts` - A `HashMap<LockedPubkey, usize>` that tracks the set of accounts needed to execute the
-current set of transactions scheduled/sent to banking threads. Accounts are added to this set
-*before* being sent to BankingStage threads. The `usize` represents a refcount, because multiple read
-accounts could be grabbed.`LockedPubkey` is defined as:
+1. `pending_transactions` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions that are not known to be blocked.
+2. `transactions_by_account` - A map `HashMap<Pubkey, AccountTransactionQueue>` that tracks all pending and blocked transactions by account key.
+   The `AccountTransactionQueue` is defined as:
+   ```
+   class AccountTransactionQueue {
+       /// Tree of read transactions on the account ordered by fee-priority
+       pub reads: RBTree<Transaction>,
+       /// Tree of write transactions on the account ordered by fee-priority
+       pub writes: RBTree<Trasaction>,
+       /// The type, count, and minimum priority transaction that is currently scheduled for execution
+       pub scheduled_lock: Option<AccountLock>,
+   }
+   ```
+   and `AccountLock` is defined as:
+   ```
+       enum AccountLock {
+           Read(usize, Transaction),
+           Write(Transaction),
+       }
+   ```
+   where the `Transaction` stored in the `AccountLock` will be the minimum priority transaction currently scheduled. The `Read` variant will also store the number
+   of read transactions.
+3. `blocked_transactions` - A map `HashMap<Signature, Vec<Transaction>>` that maps from the blocking transaction's `Signature` to the transactions it is blocking.
```

Comment:
what data structure does this go in?

Reply:
these 3 components will all be on the Scheduler